### DISCUSSION TITLE
Migration to C++14 (or later)

### DISCUSSION THREAD

**Discussion by e-kwsm** [rdkit/rdkit#4165] [Development]
At present rdkit uses C++11:
https://github.com/rdkit/rdkit/blob/f5a54af475cc38a82b2c69a8a44193a222fe92fc/CMakeLists.txt#L14

I’m implementing a reader for CML, and it would be easier to do it if I could use C++14.

Is there any plan to migrate to C++14 (or C++17)?

**Comment by greglandrum** [comment]
@e-kwsm : C++17 is problematic since it makes supporting some older systems tricky, but I think it's safe to require C++14.

**Reply by e-kwsm** [reply]
I see. Thanks.

**Reply by deadlyvices** [reply]
I'd endorse writing a CML parser. We'd like to use RDKit if we could in https://chem4word.co.uk. We make heavy use of CML.
